File: lib/primitives.ts

import casual from 'casual';

export const Primitives: Record<string, Function> = {
	country: function () { return casual.country; },
	city: function () { return casual.city; },
	zip: function () { return casual.zip(); },
	street: function () { return casual.street; },
	address: function () { return casual.address; },
	address1: function () { return casual.address1; },
	address2: function () { return casual.address2; },
	state: function () { return casual.state; },
	state_abbr: function () { return casual.state_abbr; },
	latitude: function () { return casual.latitude; },
	longitude: function () { return casual.longitude; },
	sentence: function () { return casual.sentence; },
	sentences: function (n: number) { return casual.sentences(n); },

	title: function () { return casual.title; },
	text: function () { return casual.text; },
	description: function () { return casual.description; },
	short_description: function () { return casual.short_description; },
	string: function () { return casual.string; },
	word: function () { return casual.word; },
	words: function (n: number) { return casual.words(n); },
	array_of_words: function (n: number) { return casual.array_of_words(n); },

	ip: function () { return casual.ip; },
	domain: function () { return casual.domain; },
	url: function () { return casual.url; },
	email: function () { return casual.email; },
	user_agent: function () { return casual.user_agent; },

	name: function () { return casual.name; },
	username: function () { return casual.username; },
	first_name: function () { return casual.first_name; },
	last_name: function () { return casual.last_name; },
	full_name: function () { return casual.full_name; },
	password: function () { return casual.password; },
	name_prefix: function () { return casual.name_prefix; },
	name_suffix: function () { return casual.name_suffix; },
	company_name: function () { return casual.company_name; },
	company_suffix: function () { return casual.company_suffix; },
	catch_phrase: function () { return casual.catch_phrase; },
	phone: function () { return casual.phone; },

	random: function () { return casual.random; },
	integer: function (args: { from: number, to: number }) { return casual.integer(args.from, args.to); },
	double: function (args: { from: number, to: number }) { return casual.double(args.from, args.to); },
	array_of_digits: function (n: number) { return casual.array_of_digits(n); },
	array_of_integers: function (n: number) { return casual.array_of_integers(n); },
	array_of_doubles: function (n: number) { return casual.array_of_doubles(n); },
	coin_flip: function () { return casual.coin_flip; },

	unix_time: function () { return casual.unix_time; },
	moment: function () { return casual.moment; },
	date: function (format: string) { return casual.date(format); },
	time: function (format: string) { return casual.time(format); },
	century: function () { return casual.century; },
	am_pm: function () { return casual.am_pm; },
	day_of_year: function () { return casual.day_of_year; },
	day_of_month: function () { return casual.day_of_month; },
	day_of_week: function () { return casual.day_of_week; },
	month_number: function () { return casual.month_number; },
	month_name: function () { return casual.month_name; },
	year: function () { return casual.year; },
	timezone: function () { return casual.timezone; },

	card_type: function () { return casual.card_type; },
	card_number: function (vendor: string) { return casual.card_number(vendor); },
	card_exp: function () { return casual.card_exp; },
	card_data: function () { return casual.card_data; },

	country_code: function () { return casual.country_code; },
	language_code: function () { return casual.language_code; },
	locale: function () { return casual.locale; },
	currency_code: function () { return (casual as any).currency_code; },
	currency_name: function () { return (casual as any).currency_name; },
	mime_type: function () { return casual.mime_type; },
	file_extension: function () { return casual.file_extension; },
	boolean: function () { return casual.boolean; },
	uuid: function () { return casual.uuid; },
	color_name: function () { return casual.color_name; },
	safe_color_name: function () { return casual.safe_color_name; },
	rgb_hex: function () { return casual.rgb_hex; },
	rgb_array: function () { return casual.rgb_array; },

	array_of: function (times: number, generator: any) { return casual.array_of(times, generator); },
	random_element: function (array: any[]) { return casual.random_element(array); },
	random_key: function (object: any) { return casual.random_key(object); },
	random_value: function (object: any) { return casual.random_value(object); },
	random_string: function (args: { min_length: number, max_length: number, extras: any[], exclude_digits: boolean, exclude_letters: boolean }) {
		return (casual as any).random_string({
			min_length: args.min_length, max_length: args.max_length, extras:
				args.extras, exclude_digits: args.exclude_digits, exclude_letters: args.exclude_letters
		});
	},

	register_provider: function (provider: string) { return casual.register_provider(provider); },
	numerify: function (format: string) { return casual.numerify(format); },
	randify: function (args: { format: string, extras: any[], exclude_digits: boolean, exclude_letters: boolean }) {
		return (casual as any).randify({
			format: args.format, extras: args.extras, exclude_digits: args.exclude_digits,
			exclude_letters: args.exclude_letters
		});
	},
	populate: function (format: string) { return casual.populate(format); },
	populate_one_of: function (formats: string[]) { return (casual as any).numerify(formats); },

};

File: dart/lib/primitives.dart
// ignore_for_file: non_constant_identifier_names
import 'package:faker/faker.dart';

final _faker = Faker();

class CharSets {
  static final lowercase = 'abcdefghijklmnopqrstuvwxyz';
  static final uppercase = 'ABCDEFGHIJKLMONPQESTUVWXYZ';
  static final numbers = '1234567890';
  static final special = "-._!\"`'#%&,:;<>=@{}~\$()*+/\\?[]^|";

  static getCharsSet({
    bool uppercase = true,
    bool lowercase = true,
    bool numbers = true,
    bool special = true,
  }) {
    var set = '';
    if (uppercase) set += CharSets.uppercase;
    if (lowercase) set += CharSets.lowercase;
    if (numbers) set += CharSets.numbers;
    if (special) set += CharSets.special;
    return set;
  }
}

final Primitives = {
  "zipCode": _faker.address.zipCode,
  "city": _faker.address.city,
  "city_prefix": _faker.address.cityPrefix,
  "city_suffix": _faker.address.citySuffix,
  "street_name": _faker.address.streetName,
  "street_address": _faker.address.streetAddress,
  "street_suffix": _faker.address.streetSuffix,
  "building_number": _faker.address.buildingNumber,
  "neighborhood": _faker.address.neighborhood,
  "state": _faker.address.state,
  "state_abbreviation": _faker.address.stateAbbreviation,
  "state_as_map": _faker.address.stateAsMap,
  "country": _faker.address.country,
  "country_code": _faker.address.countryCode,
  "continent": _faker.address.continent,
  "conference_name": _faker.conference.name,
  "color": _faker.color.color,
  "common_color": _faker.color.commonColor,
  "company_name": _faker.company.name,
  "company_position": _faker.company.position,
  "company_suffix": _faker.company.suffix,
  "currency_code": _faker.currency.code,
  "currency_name": _faker.currency.name,
  "restaurant": _faker.food.restaurant,
  "cuisine": _faker.food.cuisine,
  "dish": _faker.food.dish,
  "longitude": _faker.geo.longitude,
  "latitude": _faker.geo.latitude,
  "image": _faker.image.image,
  "email": _faker.internet.email,
  "disposable_email": _faker.internet.disposableEmail,
  "free_email": _faker.internet.freeEmail,
  "safe_email": _faker.internet.safeEmail,
  "user_name": _faker.internet.userName,
  "domain_name": _faker.internet.domainName,
  "domain_word": _faker.internet.domainWord,
  "uri": _faker.internet.uri,
  "http_url": _faker.internet.httpUrl,
  "https_url": _faker.internet.httpsUrl,
  "ipv4": _faker.internet.ipv4Address,
  "ipv6": _faker.internet.ipv6Address,
  "mac": _faker.internet.macAddress,
  "password": _faker.internet.password,
  "user_agent": _faker.internet.userAgent,
  "guid": _faker.guid.guid,
  "job_title": _faker.job.title,
  "word": ({
    int min_length = 0,
    int max_length = 10,
    String charset = '',
    bool uppercase = true,
    bool lowercase = true,
  }) {
    var length = random.integer(max_length, min: min_length);
    return random.fromCharSet(
        charset.isNotEmpty
            ? charset
            : CharSets.getCharsSet(uppercase: uppercase, lowercase: lowercase),
        length);
  },
  "words": ({
    int count = 3,
    int word_length = 5,
    String charset = '',
    uppercase = true,
    lowercase = true,
  }) {
    var words = [];
    var randomCount = random.integer(count);
    List.generate(randomCount + 1, (__) {
      words.add(random.fromCharSet(
          charset.isNotEmpty
              ? charset
              : CharSets.getCharsSet(
                  uppercase: uppercase, lowercase: lowercase),
          random.integer(word_length)));
    });
    return words.join(" ");
  },
  "sentence": _faker.lorem.sentence,
  "sentences": ({int count = 3}) => _faker.lorem.sentences(count),
  "name": _faker.person.name,
  "first_name": _faker.person.firstName,
  "last_name": _faker.person.lastName,
  "prefix": _faker.person.prefix,
  "phone_number": ({String pattern = ''}) => random.fromPattern([pattern]),
  "vehicle_make": _faker.vehicle.make,
  "vehicle_model": _faker.vehicle.model,
  "vehicle_year": _faker.vehicle.year,
  "vehicle_year_make_model": _faker.vehicle.yearMakeModel,
  "vehicle_color_year_makemodel": _faker.vehicle.colorYearMakeModel,
  "vin": _faker.vehicle.vin,
  "month": _faker.date.month,
  "year": _faker.date.year,
  "date": () => random.integer(31, min: 1),
  "day": () => random.element([
        'saturday',
        'sunday',
        'monday',
        'tuesday',
        'wednesay',
        'thurdsay',
        'friday'
      ]),
  "time": _faker.date.time,
  "just_time": _faker.date.justTime,
  "random_element_from_list": random.element,
  "random_key": random.mapElementKey,
  "random_value": random.mapElementValue,
  "boolean": random.boolean,
  "decimal": ({num scale = 1, num min = 0}) => random.decimal,
  "number_of_length": ({int length = 10}) => random.numberOfLength,
  "number": ({int min = 0, int max = 10}) => random.integer(max, min: min),
  "numbers_from_pattern": ({String pattern = "####"}) =>
      random.fromPattern([pattern]),
  "string_from_pattern": ({
    String charset = '',
    String pattern = "####",
    bool uppercase = true,
    bool lowercase = true,
    bool numbers = true,
    bool special = true,
  }) {
    return pattern.splitMapJoin(
      '#',
      onMatch: (_) => random.fromCharSet(
          charset.isNotEmpty
              ? charset
              : CharSets.getCharsSet(
                  uppercase: uppercase,
                  lowercase: lowercase,
                  numbers: numbers,
                  special: special,
                ),
          1),
    );
  },
  "from_charset": ({
    String charset = '',
    bool uppercase = true,
    bool lowercase = true,
    bool numbers = true,
    bool special = true,
    int length = 10,
  }) =>
      random.fromCharSet(
        charset.isNotEmpty
            ? charset
            : CharSets.getCharsSet(
                uppercase: uppercase,
                lowercase: lowercase,
                numbers: numbers,
                special: special,
              ),
        length,
      )
};
